### PR TITLE
Bump rsa to 0.7

### PR DESCRIPTION
**Pull request by nox**

### CODE REVIEW

**Comment by nox** [comment]
Feel free to reject this PR, I made it to try to dedupe crates in my tree.

File: src/algorithms/rsa.rs [hunk 1]
```diff
@@ -114,7 +114,7 @@ impl RSAKeyPair {
         self.rsa_sk
             .to_pkcs8_der()
             .map_err(Into::into)
-            .map(|x| x.as_ref().to_vec())
+            .map(|x| std::mem::take(x.to_bytes().as_mut()))
```
Comment (nox) [inline src/algorithms/rsa.rs:117]: RSA now returns a `Zeroizing<Vec<u8>>` and I found no easier way to unwrap it.

Comment (jedisct1) [reply src/algorithms/rsa.rs:117]: Congrats for figuring out that magic :)

File: Cargo.toml [hunk 2]
```diff
@@ -25,10 +25,11 @@ k256 = { version = "0.11.6", features = ["ecdsa", "std", "pkcs8", "pem"] }
 p256 = { version = "0.11.1", features = ["ecdsa", "std", "pkcs8", "pem"] }
 p384 = { version = "0.11.2", features = ["ecdsa", "std", "pkcs8", "pem"] }
 rand = "0.8.5"
-rsa = "0.6.1"
+rsa = "0.7"
 serde = { version = "1.0.145", features = ["derive"] }
 serde_json = "1.0.86"
-spki05 = { package = "spki", version = "0.5.4" }                           # Use an old version, required by the rsa crate, but that crate doesn't reexport it
+sha2 = { version = "0.10.6", features = ["oid"] }
```
Comment (jedisct1) [inline Cargo.toml:31]: Err... now we have more code duplication.

`hmac-sha256` and `hmac-sha512` already provide SHA2 implementations.

File: src/algorithms/rsa.rs [hunk 3]
```diff
@@ -259,7 +259,7 @@ impl RSAKeyPairLike for RS256KeyPair {
     }
 
     fn padding_scheme(&self) -> rsa::PaddingScheme {
-        rsa::PaddingScheme::new_pkcs1v15_sign(Some(rsa::Hash::SHA2_256))
+        rsa::PaddingScheme::new_pkcs1v15_sign::<sha2::Sha256>()
```
Comment (jedisct1) [inline src/algorithms/rsa.rs:262]: Does that work?

I'm a little bit scared by the fact that an object that depends on randomness doesn't get a RNG any more.

Maybe it's reusing the one from `sign_blinded()` but that would be very counterintuitive.

Before bumping the dependency, we need to make sure that the behavior stays the same and that it doesn't introduce a vulnerability.

File: Cargo.toml (hunk 2 above)
Comment (nox) [reply Cargo.toml:31]: We could either make your crates depend on sha2, but then they wouldn’t be self-contained anymore, or we could expose stuff in them that satisfy the trait bounds required by the rsa crate. IIRC all the traits come from digest and your crates already depend on it.

File: src/algorithms/rsa.rs (hunk 3 above)
Comment (nox) [reply src/algorithms/rsa.rs:262]: I had the same worry as you, mostly opened the PR to start the discussion, I fully agree with you that we need to be sure what the change was about before we merge this.

File: Cargo.toml (hunk 2 above)
Comment (jedisct1) [reply Cargo.toml:31]: The current version of the `hmac-*` crates implement the `AssociatedOid` trait, so we shouldn't need `sha2`, just a version bump in `Cargo.toml`.

File: src/algorithms/rsa.rs (hunk 3 above)
Comment (jedisct1) [reply src/algorithms/rsa.rs:262]: Glancing at the code, it looks like the RNG is implicitly passed over from the signing operation to the PSS object. So , the behavior should be the same.

File: Cargo.toml (hunk 2 above)
Comment (jedisct1) [reply Cargo.toml]: Done :)

**Comment by nox** [comment]
Thanks for the swift merge! Any chance we could have a new release?

**Pull request merged**